src/store/migrations/db.js

```diff
@@ -0,0 +1,185 @@
+// @flow strict
+
+const DB_NAME: string = 'localforage'
+const STORE_NAME: string = 'keyvaluepairs'
+const VERSION_KEY: string = 'persist:jwallet-web-version'
+
+function openDB(): Promise<IDBDatabase> {
+  return new Promise((resolve, reject) => {
+    const openRequest: IDBOpenDBRequest = window.indexedDB.open(DB_NAME)
+
+    /* eslint-disable fp/no-mutation */
+    openRequest.onsuccess = function openDBSuccess() {
+      // $FlowFixMe
+      resolve(openRequest.result)
+    }
+
+    openRequest.onerror = function openDBError() {
+      reject(new Error(`Database "${DB_NAME}" open error`))
+    }
+    /* eslint-enable fp/no-mutation */
+  })
+}
+
+export function initTransaction(): Promise<IDBTransaction> {
+  return openDB().then((db: IDBDatabase) => {
+    const transaction: IDBTransaction = db.transaction(STORE_NAME, 'readwrite')
+
+    /* eslint-disable fp/no-mutation */
+    transaction.onerror = function transactionError() {
+      throw new Error('Transaction failed')
+    }
+
+    transaction.oncomplete = function transactionComplete() {
+      // console.log('Transaction completed')
+    }
+
+    return Promise.resolve(transaction)
+    /* eslint-enable fp/no-mutation */
+  })
+}
+
+export function getStoreVersion(): Promise<number> {
+  return initTransaction().then((transaction: IDBTransaction) => new Promise((resolve, reject) => {
+    const objectStore: IDBObjectStore = transaction.objectStore(STORE_NAME)
+    const objectStoreRequest: IDBRequest = objectStore.get(VERSION_KEY)
+
+    /* eslint-disable fp/no-mutation */
+    objectStoreRequest.onsuccess = function objectStoreSuccess() {
+      resolve(parseInt(objectStoreRequest.result, 10) || 0)
+    }
+
+    objectStoreRequest.onerror = function objectStoreError() {
+      reject(new Error('Request failed'))
+    }
+    /* eslint-enable fp/no-mutation */
+  }))
+}
+
+export function setStoreVersion(
+  version: number,
+  transaction: IDBTransaction,
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const objectStore: IDBObjectStore = transaction.objectStore(STORE_NAME)
+    const objectStoreRequest: IDBRequest = objectStore.put(version.toString(), VERSION_KEY)
+
+    /* eslint-disable fp/no-mutation */
+    objectStoreRequest.onsuccess = function objectStoreSuccess() {
+      resolve()
+    }
+
+    objectStoreRequest.onerror = function objectStoreError() {
+      reject(new Error('Request failed'))
+    }
+    /* eslint-enable fp/no-mutation */
+  })
+}
+
+export function getStoreData(key: string): Promise<Object> {
```

Comment:
If you are interconnecting these functions with specific store format (you are suggesting that values are in JSON format), the file should not be called `db` – because these are not general functions for working with db, but specific functions for working with persist db.

There are also common functions here, like db opening, getting store version, initialising the transaction, that do not depend on actual storage structure and can be used between migration. Those are better to separate into other file for future reuse.

Reply:
But `persist` container will be necessary in future as well. We need to separate data that must be persisted and data that mustn't. `persist` container is good way for this.

Reply:
Basically, all the data you have in storage is already persisted.

Reply:
But in some reducers some data shouldn't be persisted. For that case `persist` container is ok.

Reply:
Ah, ok, this container inside indexedDB is needed only when we are using redux-persist.